File: pkg/types/job.go
package types

import "encoding/json"

type Job struct {
	Id            string `json: "id"`
	ProjectId     string `json: "projectId"`
	AgentId       string `json: "agentId"`
	EnvironmentId string `json: "environmentId"`
	TaskForceId   string `json: "taskForceId"`
	CreatedAt     string `json: "createdAt"`
	Status        string `json: "status"`
}

func (it *Job) AsJson() string {
	b, err := json.Marshal(it)
	if err != nil {
		panic("Unable to marshal object")
	}
	return string(b)
}

File: core/src/main/java/org/robotframework/roc/core/job/Job.java
package org.robotframework.roc.core.job;

import lombok.Data;
import org.springframework.data.redis.core.RedisHash;

import java.io.Serializable;
import java.util.Date;

@Data
@RedisHash("Job")
public class Job implements Serializable {

    private String id;

    private String projectId;
    private String agentId;
    private String environmentId;
    private String taskForceId;
    
    private String name;
    private JobStatus status;

    private Date createdAt;
    private Date startedAt = null;
    private Date finishedAt = null;

}
